src/coredump/coredump.c

```diff
@@ -1063,9 +1066,24 @@ static int context_parse_iovw(Context *context, struct iovec_wrapper *iovw) {
                 if (context->pidref.pid != parsed_pid)
                         return log_error_errno(r, "Passed PID " PID_FMT " does not match passed " PID_FMT ": %m", parsed_pid, context->pidref.pid);
         } else {
-                r = pidref_set_pid(&context->pidref, parsed_pid);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to initialize pidref from pid " PID_FMT ": %m", parsed_pid);
+                if (!isempty(context->meta[META_ARGV_PIDFD]))  {
+                        int fd;
+
+                        r = safe_atoi(context->meta[META_ARGV_PIDFD], &fd);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse PIDFD \"%s\": %m", context->meta[META_ARGV_PIDFD]);
+
+                        r = pidref_set_pidfd_consume(&context->pidref, fd);
```

Comment:
it doesn't matter too much, but you take possession of the fd here, half-way, i.e. there are a myriad of places this function can still fail later. given this is not performance sensitive, i'd kinda prefer if we'd initialize the pidref from a duplicated fd, and keep the main fd open during the rest of the function, and close it only once we determined the function cannot fail anymore.

as i understand the kernel will put the pidfd always at fd index 3, but it doesn't communicate that as api, hence I think that would be nicer already to move the fd out of any risky area (given that otherwise coredump handlers have uninitialized fd1,2)